apps/gallery/test/marionette/lib/gallery.js

```diff
@@ -310,6 +386,45 @@ Gallery.prototype = {
     }.bind(this));
   },
 
+  switchToGalleryFrame: function() {
+    var galleryFrame = this.client.findElement('iframe[src*="' +
+                                          Gallery.ORIGIN + '"]');
+    this.waitFor(galleryFrame);
+    this.client.switchToFrame(galleryFrame);
+  },
+
+  tapFirstThumbnail: function() {
+    this.waitFor(Gallery.Selector.thumbnail);
+    var thumbnails = this.thumbnails;
+    this.waitFor(thumbnails[0]);
+    thumbnails[0].click();
+  },
+
+  selectImage: function() {
+    this.switchToGalleryFrame();
+    this.tapFirstThumbnail();
+  },
```

Comment:
I think that this function starts off assuming we're in the system app, and ends up in the gallery app.  If so, then I'd suggest that it does not belong in this library, but should be a utility function in the test itself.  In the test, I'd expect to see something like:

```
system.switchToGallery()
gallery.tapFirstThumbnail()
```